asset/src/__lib/schema.ts

```diff
@@ -0,0 +1,29 @@
+import { getTypeOf, isString, isNotNil } from '@terascope/job-components';
+import { DEFAULT_API_NAME as ES_DEFAULT_API_NAME } from '../elasticsearch_reader_api/interfaces';
+import { DEFAULT_API_NAME as SPACES_DEFAULT_API_NAME } from '../spaces_reader_api/interfaces';
+
+export const elasticOpSchema = {
```

Comment:
A few things:

- `elasticOpSchema` is the same as `spacesOpSchema` (even the error messages say `spaces_reader_api`)
- The api name doesn't have to include the default api name because a developer may create a different reader api with the same interface and that should be totally valid to use.
- I think just setting this `format: 'optional_String'` is totally fine